rotkehlchen/db/filtering.py

```diff
@@ -1062,3 +1082,144 @@ def make(
+        filter_query = cast('NFTFilterQuery', filter_query)
+        filters: List[DBFilter] = []
+        if owner_addresses is not None:
+            filters.append(DBMultiStringFilter(
+                and_op=True,
+                column='owner_address',
+                values=owner_addresses,
+            ))
+        if name is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='name',
+                search_string=name,
+            ))
+        if collection_name is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='collection_name',
+                search_string=collection_name,
+            ))
+        if ignored_assets_filter_params is not None:
+            filters.append(DBMultiStringFilter(
+                and_op=True,
+                column='identifier',
+                operator=ignored_assets_filter_params[0],
+                values=ignored_assets_filter_params[1],
+            ))
+        filter_query.filters = filters
+        return filter_query
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class MultiTableFilterQuery():
+    and_op: bool
+    filters: List[Tuple[DBFilter, str]]  # filter + table name for which to use it.
+
+    def prepare(self, target_table_name: str) -> Tuple[str, List[Any]]:
+        query_parts = []
+        bindings = []
+        filterstrings = []
+
+        for (fil, table_name) in self.filters:
+            if table_name != target_table_name:
+                continue
+
+            filters, single_bindings = fil.prepare()
+            if len(filters) == 0:
+                continue
+
+            operator = ' AND ' if fil.and_op else ' OR '
+            filterstrings.append(f'({operator.join(filters)})')
+            bindings.extend(single_bindings)
+
+        if len(filterstrings) != 0:
+            operator = ' AND ' if self.and_op else ' OR '
+            filter_query = f'WHERE {operator.join(filterstrings)}'
+            query_parts.append(filter_query)
+
+        return ' '.join(query_parts), bindings
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class LevensteinFilterQuery(MultiTableFilterQuery):
+    substring_search: str
+
+    @classmethod
+    def make(
+            cls,
+            and_op: bool = True,
+            substring_search: str = '',  # substring is always required for levenstein
+            chain_id: Optional[ChainID] = None,
+    ) -> 'LevensteinFilterQuery':
+        filter_query = LevensteinFilterQuery(
+            and_op=and_op,
+            filters=[],
+            substring_search=substring_search,
+        )
+        filters: List[Tuple[DBFilter, str]] = []  # filter + table name for which to use it.
+
+        name_filter = DBSubStringFilter(
+            and_op=True,
+            field='name',
+            search_string=substring_search,
+        )
+
```

Comment:
```suggestion
        )
```

Reply:
I think I understood that you meant to remove an empty line, but the comment looks funny :)
![image](https://user-images.githubusercontent.com/63492346/201492854-8753e34e-7d0c-4961-97fb-beaad85ace15.png)